src/System.Security.Cryptography.Primitives/src/System/Security/Cryptography/CryptoStream.cs

```diff
@@ -14,27 +14,27 @@ public class CryptoStream : Stream, IDisposable
     {
         // Member variables
         private readonly Stream _stream;
-        private readonly ICryptoTransform _transform;
-        private byte[] _inputBuffer;  // read from _stream before _Transform
+        private readonly ICryptoTransform? _transform;
+        private byte[]? _inputBuffer;  // read from _stream before _Transform
         private int _inputBufferIndex;
         private int _inputBlockSize;
-        private byte[] _outputBuffer; // buffered output of _Transform
+        private byte[]? _outputBuffer; // buffered output of _Transform
         private int _outputBufferIndex;
         private int _outputBlockSize;
         private bool _canRead;
         private bool _canWrite;
         private bool _finalBlockTransformed;
-        private SemaphoreSlim _lazyAsyncActiveSemaphore;
+        private SemaphoreSlim? _lazyAsyncActiveSemaphore;
         private readonly bool _leaveOpen;
 
         // Constructors
 
-        public CryptoStream(Stream stream, ICryptoTransform transform, CryptoStreamMode mode)
+        public CryptoStream(Stream stream, ICryptoTransform? transform, CryptoStreamMode mode)
             : this(stream, transform, mode, false)
         {
         }
 
-        public CryptoStream(Stream stream, ICryptoTransform transform, CryptoStreamMode mode, bool leaveOpen)
+        public CryptoStream(Stream stream, ICryptoTransform? transform, CryptoStreamMode mode, bool leaveOpen)
```

Comment:
(Repeat from the ref) While technically null can be passed in here without the ctor throwing, the object is in an invalid state.  So I'd mark these as non-nullable (and potentially add the ArgumentNullException).